maven-compat/src/main/java/org/apache/maven/repository/legacy/DefaultUpdateCheckManager.java

```diff
@@ -350,60 +350,29 @@ private Properties read( File touchfile )
 
         synchronized ( touchfile.getAbsolutePath().intern() )
         {
-            FileInputStream in = null;
-            FileLock lock = null;
-
             try
             {
                 Properties props = new Properties();
 
-                in = new FileInputStream( touchfile );
-                lock = in.getChannel().lock( 0, Long.MAX_VALUE, true );
-
-                getLogger().debug( "Reading resolution-state from: " + touchfile );
-                props.load( in );
-
-                lock.release();
-                lock = null;
+                try ( FileInputStream in = new FileInputStream( touchfile ) )
+                {
+                    try ( FileLock lock = in.getChannel().lock( 0, Long.MAX_VALUE, true ) )
+                    {
+                        getLogger().debug( "Reading resolution-state from: " + touchfile );
+                        props.load( in );
 
-                in.close();
-                in = null;
+                        lock.release();
```

Comment:
`lock.release();` is not needed because `FileLock::close` calls `FileLock::release`.
`FileLock::close` will be called automatically when `try-with-resource` block ends.